feature-libs/cart/base/root/models/cart.model.ts

```diff
@@ -116,6 +117,32 @@ export interface CartItemComponentOptions {
   addToCartString?: string;
   cartType?: CartType;
 }
+/**
+ * A key that identifies an 'abstract order', that in OO terms can be understood
+ * as a common super class of cart, saved cart, order and quote.
+ */
+export interface AbstractOrderKey {
+  type: AbstractOrderType;
+  /**
+   * Identifier of an abstract order. The attribute is optional because for the active cart the id
+   * does not need to be present.
+   * For the other types (SavedCart, Order and Quote) the id is mandatory in order to identify
+   * the document.
+   */
+  id?: string;
+}
```

Comment:
[MEDIUM]
### 1. `id` should not be optional
The more I think about it, the more I can't get rid of the impression that the declaring in Typescript the property `id` as  _optional_ is confusing. In reality, it's is not optional. It's always **either** present or **either** absent, **depending strictly** on the abstract order _type_ (e.g. whether it's the active cart or something else).

IMHO a more precise and predictable structure would be to NOT separate the properties `type` and `id`, but rather create an union type of strict and specific objects **either** having `id` or **either** not. See suggestion:

```suggestion
/**
 * A key that identifies an 'abstract order', that in OO terms can be understood
 * as a common super class of cart, saved cart, order and quote.
 *
 * The attribute `id` is absent for the ActiveCart type, because there exists only one active cart.
 * which can be identified as the `current` one.
 * For the other types (SavedCart, Order and Quote) the attirbute `id` is present in order to identify
 * the abstract order.
 */
export type AbstractOrderKey =
  | { type: AbstractOrderType.CART }
  | { type: AbstractOrderType.ORDER; id: string }
  | { type: AbstractOrderType.QUOTE; id: string }
  | { type: AbstractOrderType.SAVED_CART; id: string };
```

Such a type allows Typescript for inferring whether `id` is present or not, after checking in our source code the `type`. For example:
```ts
if (abstractOrderKey.type !== AbstractOrderType.CART) {
  // Typescript infers that `abstractOrderKey.id` is present
} else{
  // Typescript infers that `abstractOrderKey.id` is absent
}
```
(for more, see [stackblitz example](https://stackblitz.com/edit/typescript-14apac?file=index.ts))

### 2. `type$` and `id$` should be in one observable, because they are inseparable
The `type` and `id` never change in separation, independently form each other. Therefore IMHO in the `AbstractOrderContext` we shouldn't have 2 separate observables (which suggest that `type` and `id` could change independently). Instead IMHO we should expose just one observable with the `AbstractOrderKey` object. 
Btw. it will also help avoid the need to create `abstractOrderData$` in `configure-cart-entry.component.ts`:
```diff
-  abstractOrderData$: Observable<AbstractOrderKey> = this.abstractOrderContext
-    ? combineLatest([
-        this.abstractOrderContext.id$,
-        this.abstractOrderContext.type$,
-      ]).pipe(map(([id, type]) => ({ id, type })))
+  abstractOrderKey$: Observable<AbstractOrderKey> = this.abstractOrderContext
+    ? this.abstractOrderContext.abstractOrderKey$
     : of({ type: AbstractOrderType.CART });
```